### PR TITLE
feat(csi-liveness-probe): longhorn-csi-plugin

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/3907

### CODE REVIEW

File: csi/deployment.go [hunk 1]
```diff
@@ -23,6 +25,7 @@ const (
 	DefaultCSIResizerImage             = "longhornio/csi-resizer:v1.3.0"
 	DefaultCSISnapshotterImage         = "longhornio/csi-snapshotter:v5.0.1"
 	DefaultCSINodeDriverRegistrarImage = "longhornio/csi-node-driver-registrar:v2.5.0"
+	DefaultCSILivenessProbeImage       = "registry.k8s.io/sig-storage/livenessprobe:v2.8.0"
```
Comment (c3y1huang) [inline csi/deployment.go:28]: TODO: update after https://github.com/longhorn/longhorn/issues/4945.

File: csi/deployment.go [hunk 2]
```diff
@@ -387,6 +391,20 @@ func NewPluginDeployment(namespace, serviceAccount, nodeDriverRegistrarImage, ma
 								},
 							},
 						},
+						{
+							Name:            "longhorn-liveness-probe",
+							ImagePullPolicy: imagePullPolicy,
+							Image:           livenessProbeImage,
+							Args: []string{
+								fmt.Sprintf("--csi-address=%s", GetInContainerCSISocketFilePath()),
```
Comment (innobead) [inline csi/deployment.go:399]: No verbose? Just have a try to see what level is much suitable. 

ref: https://github.com/kubernetes-csi/livenessprobe/blob/master/deployment/kubernetes/livenessprobe-sidecar.yaml#L51

**Review by innobead — COMMENTED** [review]
LGTM, excetp one comment.

File: csi/deployment.go (hunk 2 above)
Comment (c3y1huang) [reply csi/deployment.go:399]: Yes, we should set verbose, I think the default is set to `v=2`. 
`v=5` output the [probe request](https://github.com/kubernetes-csi/livenessprobe/blob/master/cmd/livenessprobe/main.go#L69) and the [connections](https://github.com/kubernetes-csi/livenessprobe/blob/master/vendor/github.com/kubernetes-csi/csi-lib-utils/connection/connection.go#L183-L187), so set `v=4` should be at the debug level and looks right to me.

**Review by innobead — APPROVED** [review]

File: csi/deployment.go (hunk 2 above)
Comment (c3y1huang) [reply csi/deployment.go:400]: Ok, I think v=4 is still too much. Need to make it less verbose.
```
[longhorn-csi-plugin-hf5vb longhorn-csi-plugin] time="2022-11-29T03:27:43Z" level=info msg="Probe: rsp: {}"
[longhorn-csi-plugin-rsvwr longhorn-csi-plugin] time="2022-11-29T03:27:43Z" level=info msg="Probe: req: {}"
```

**Pull request merged**